# AirBnB 
The project entails developing a machine learning-based solution for Airbnb to suggest appropriate listing prices to property owners in Antwerp, Belgium. This requires analyzing datasets from the company's data warehouse, including listings, calendar, hosts, and reviews information. The tasks involve data understanding and feature creation, data quality checks, variable profiling, and modeling to predict listing prices using various regression techniques. The goal is to identify significant predictors affecting prices and ensure the data's readiness for analysis through cleaning and transformation processes.

In [1]:
import pandas as pd

In [2]:
# Load the provided datasets
calendar_df = pd.read_csv('calendar.csv')
hosts_df = pd.read_csv('hosts.csv')
listings_df = pd.read_csv('listings.csv')
reviews_df = pd.read_csv('reviews.csv')

In [15]:
# Display the first few rows of each dataframe to understand their structure
print("Calendar : ", calendar_df.shape)
display(calendar_df.head())
print("Hosts : ", hosts_df.shape)
display(hosts_df.head())

Calendar :  (319192, 8)


calender_id  listing_id                        date  available  price  \
0            1    40334325  2022-08-03 00:00:00.000000          0   56.0   
1            2    22742449  2022-11-13 00:00:00.000000          1   95.0   
2            3    34621717  2022-04-17 00:00:00.000000          0   75.0   
3            4    38281744  2022-01-31 00:00:00.000000          1  150.0   
4            5    18835003  2022-05-21 00:00:00.000000          0  100.0   

   adjusted_price  minimum_nights  maximum_nights  
0            56.0               3               5  
1            95.0               2              99  
2            75.0               2            1125  
3           150.0               1            1000  
4           100.0               2            1125

Hosts :  (1111, 5)


host_id host_name                  host_since  \
0   234077     Karin  2010-09-14 00:00:00.000000   
1   334804       Ann  2011-01-04 00:00:00.000000   
2   413052   Valérie  2011-02-27 00:00:00.000000   
3   452791   Tatiana  2011-03-20 00:00:00.000000   
4   462975       Els  2011-03-25 00:00:00.000000   

                      host_location  \
0        Antwerp, Flanders, Belgium   
1  Antwerp, Flemish Region, Belgium   
2        Antwerp, Flanders, Belgium   
3        Antwerp, Flanders, Belgium   
4         Edegem, Flanders, Belgium   

                                          host_about  
0  Ever since my childhood I dreamt of having my ...  
1  Ciao, \r\nlooking forward to meet you!\r\nI lo...  
2                                                NaN  
3  World traveler with a penchant for adrenaline ...  
4  I studied languages and cultural anthropology ...

In [12]:
print("listings : ",listings_df.shape)
display(listings_df.head(2))
print("Reviews : ",reviews_df.shape)
display(reviews_df.head())

listings :  (1749, 14)


listing_id                          listing_url  \
0       50904   https://www.airbnb.com/rooms/50904   
1      116134  https://www.airbnb.com/rooms/116134   

                                            name  \
0  aplace/antwerp: cosy suite - fashion district   
1                  Spacious apartment nearby Mas   

                                         description   latitude  longitude  \
0  Decorated in a vintage style combined with a f...  51.218575   4.398631   
1  Enjoy your stay at our 4 person apartment in t...  51.230510   4.405930   

            property_type        room_type  accomodates  bathrooms_text  \
0  Room in boutique hotel       Hotel room            2  1 private bath   
1      Entire rental unit  Entire home/apt            4       2.5 baths   

   bedrooms  beds                                          amenities  host_id  
0       1.0   1.0  ["Kitchen", "Smoke alarm", "Hair dryer", "Keyp...   234077  
1       2.0   2.0  ["Refrigerator", "Elevator", "Paid street park...   586942

Reviews :  (62987, 6)


review_id  listing_id                        date  reviewer_id  \
0          1       50904  2015-05-06 00:00:00.000000     19482395   
1          2       50904  2021-10-10 00:00:00.000000       333559   
2          3      116134  2012-03-05 00:00:00.000000       928644   
3          4      116134  2012-05-25 00:00:00.000000       231288   
4          5      116134  2013-09-03 00:00:00.000000      7984251   

  reviewer_name                                           comments  
0         Jihae  Karin’s “Aplace” is absolutely beautiful and c...  
1        Emilie  Karin is a wonderful host, she was really help...  
2      Aurélien  Amazing flat, really close from the MAS Musem,...  
3          Gail  This is a well equipped, very comfortable apar...  
4        Marcel  This is a very nice appartement. We really lik...

Here's an overview of the datasets provided:

### Calendar Dataset
- Shape: 319,192 rows × 8 columns
- Contains chronological data on listings, including availability, price, adjusted price, and booking constraints (minimum and maximum nights).

### Hosts Dataset
- Shape: 1,111 rows × 5 columns
- Information on hosts who've posted their listings, including their ID, name, registration timestamp, location, and a self-reported description.

### Listings Dataset
- Shape: 1,749 rows × 14 columns
- Detailed data on individual listings, covering aspects like ID, URL, name, description, geographical coordinates, property type, room type, accommodation capacity, number of bathrooms, bedrooms, beds, amenities, and host ID.

### Reviews Dataset
- Shape: 62,987 rows × 6 columns
- Data on reviews, including the listing ID, review ID, posting date, reviewer ID and name, and the review comment itself.